homeassistant/components/starlink/__init__.py

```diff
@@ -0,0 +1,155 @@
+"""The Starlink integration."""
+from __future__ import annotations
+
+from abc import ABC
+from datetime import timedelta
+import logging
+
+import async_timeout
+from spacex.starlink import CommunicationError, StarlinkDish
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_NAME, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA_BASE
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    CONF_DEFAULT_ADDRESS,
+    CONF_DEFAULT_NAME,
+    CONF_DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[str] = ["binary_sensor", "sensor"]
```

Comment:
Only one platform is allowed to be added when a new integration is added. This makes things easier for the member who will review this PR while still providing some functionality for the user. Please pick one. We also now have an Enum for platforms. Please use it. Thanks